client/state-db/src/pruning.rs

```diff
@@ -463,53 +368,27 @@ impl<BlockHash: Hash, Key: Hash, D: MetaDb> RefWindow<BlockHash, Key, D> {
 		// do not have this block
 		if self.is_empty() ||
 			number < self.pending() ||
```

Comment:
```suggestion
			number < self.base ||
```
more readable I think, maybe pending function can be remove, I did not check.